client/src/pages/Achieves/utils.ts

```diff
@@ -0,0 +1,168 @@
+
+    switch (achieveId) {
+      case AchievesId.FRIENDS:
+        return checkFriendAchieve(user);
+
+      case AchievesId.DISCOVERER:
+        return checkDiscovererAchieve(stats);
+
+      case AchievesId.EXPLORER:
+        return checkExplorerAchieve(stats);
+
+      case AchievesId.SHELDON:
+        if (countMemoryAttemts(stats) >= 10) {
+          return true;
+        }
+        return false;
+
+      case AchievesId.FLASH:
+        if (countMemoryAttemts(stats) >= 30) {
+          return true;
+        }
+        return false;
+
+      case AchievesId.MNEMONIST:
+        if (countMemoryAttemts(stats) >= 50) {
+          return true;
+        }
+        return false;
+
+      case AchievesId.FLOW:
+        if (countConcentrationAttemts(stats) >= 10) {
+          return true;
+        }
+        return false;
+
+      case AchievesId.RESOURCE:
+        if (countConcentrationAttemts(stats) >= 30) {
+          return true;
+        }
+        return false;
+
+      case AchievesId.MOMENT:
+        if (countConcentrationAttemts(stats) >= 50) {
+          return true;
+        }
+        return false;
+
+      case AchievesId.BATMAN:
+        if (countLogicsAttemts(stats) >= 10) {
+          return true;
+        }
+        return false;
+
+      case AchievesId.INTELLECTUAL:
+        if (countLogicsAttemts(stats) >= 30) {
+          return true;
+        }
+        return false;
+
+      case AchievesId.THINKER:
+        if (countLogicsAttemts(stats) >= 50) {
+          return true;
+        }
+        return false;
+
+      case AchievesId.WEEK:
+        if (countDaysInARow(stats) >= 7) {
+          return true;
+        }
+        return false;
+
+      case AchievesId.HABIT:
+        if (countDaysInARow(stats) >= 21) {
+          return true;
+        }
+        return false;
+
+      default:
+        return false;
+    }
+  } else {
+    return false;
+  }
+};
+
+function checkFriendAchieve(user: User) {
+  if (user.name && user.surname && user.gender && user.birthDay && user.education) {
+    return true;
+  }
+  return false;
+}
+
+function checkDiscovererAchieve(statistics: GameAttempt[]) {
+  const categories = statistics.map((item) => item.category);
+  if (
+    categories.includes(GameCategories.CONCENTRATION) &&
+    categories.includes(GameCategories.MEMORY) &&
+    categories.includes(GameCategories.LOGICS)
+  ) {
+    return true;
+  }
+  return false;
+}
+
+function checkExplorerAchieve(statistics: GameAttempt[]) {
+  let memory = 0;
+  let logics = 0;
+  let concentration = 0;
+  statistics.forEach((item) => {
+    if (item.category === GameCategories.CONCENTRATION) {
+      concentration += 1;
+    }
+    if (item.category === GameCategories.LOGICS) {
+      logics += 1;
+    }
+    if (item.category === GameCategories.MEMORY) {
+      memory += 1;
+    }
+  });
+  if (memory >= 5 && concentration >= 5 && logics >= 5) {
+    return true;
+  }
+  return false;
+}
+
+function countMemoryAttemts(statistics: GameAttempt[]) {
```

Comment:
It looks like we can reuse this code if we add one more parameter - category.

Reply:
Refactored